doc/sphinx/practical-tools/coqide.rst

```diff
@@ -5,130 +5,165 @@
+  color and the text is shown in blue and underlined.  The message text is
+  available as a tooltip on the text of the command.
+- Fail with an error.  If you're stepping through the proof line by line, the
+  error message appears in the message panel in red and the command is shown
+  in red and underlined with a pink background.  If you're in async mode,
+  described in more detail later, the message appears in the *errors panel*.
+  Double click on an entry to jump to the point of the error.  Execution
+  of commands stops unless you're in async mode.
+
+.. todo: describe the ribbon (a bit broken, though)
+
+In the previous figure :ref:`CoqIDE main screen <coqide_mainscreen>`,
 the running buffer is `Fermat.v`, all commands until
 the ``Theorem`` have been already executed, and the user tried to go
 forward executing ``Induction n``. That command failed because no such
 tactic exists (names of standard tactics are written in lowercase),
 and the failing command is underlined.
 
-Notice that the processed part of the running buffer is not editable. If
-you ever want to modify something you have to go backward using the up
-arrow tool, or even better, put the cursor where you want to go back
-and use the goto button. Unlike with `coqtop`, you should never use
-``Undo`` to go backward.
-
-There are two additional buttons for navigation within the running buffer. The
-"down" button with a line goes directly to the end; the "up" button with a line
-goes back to the beginning. The handling of errors when using the go-to-the-end
-button depends on whether Coq is running in asynchronous mode or not (see
-Chapter :ref:`asynchronousandparallelproofprocessing`). If it is not running in that mode, execution
-stops as soon as an error is found. Otherwise, execution continues, and the
-error is marked with an underline in the error foreground color, with a
-background in the error background color (pink by default). The same
-characterization of error-handling applies when running several commands using
-the "goto" button.
-
-If you ever try to execute a command that runs for a long time
-and would like to abort it before it terminates, you may
-use the interrupt button (the white cross on a red circle).
+If you're not in async mode and you modify the processed part of the buffer,
+everything after that point is undone.  Unlike `coqtop`, you should not use
+:cmd:`Undo` to go backward.
+
+.. todo drop next para, mention left and right with search
 
 There are other buttons on the CoqIDE toolbar: a button to save the running
 buffer; a button to close the current buffer (an "X"); buttons to switch among
-buffers (left and right arrows); an "information" button; and a "gears" button.
+buffers (left and right arrows); and a "gears" button.
+
+Async mode
+----------
+
+Asynchronous mode uses multiple Coq processes to process elements of the script
+in parallel.  For example, multiple theorems may be processed in parallel.  This is
```

Comment:
I would say "concurrent" rather than "parallel". Concurrent would mean that the application is processing many separate tasks at the same time, whilst parallel would indicate there is one task broken down into smaller subtasks.

Reply:
Then again maybe we made a mistake with calling it "parallel proof processing". I suppose the task depends on your point of view.

Reply:
It is parallel (when things go well :-) ).  Coq creates multiple proof worker processes that can run on different cores.  Not merely concurrent.

Reply:
Right, but my point was that they are not processing the same proof, but separate proofs in the same buffer. Minor point anyway.

Reply:
The "multiple theorems" can be processed in parallel, but individual theorems can't be split and parallelized as things are now. (Maybe different in the future).  I think that would be the most obvious understanding from the current wording.